blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxLocatorIO.java

```diff
@@ -0,0 +1,148 @@
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.querybuilder.Select;
+import com.rackspacecloud.blueflood.cache.TenantTtlProvider;
+import com.rackspacecloud.blueflood.io.CassandraModel;
+import com.rackspacecloud.blueflood.io.Instrumentation;
+import com.rackspacecloud.blueflood.io.LocatorIO;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.utils.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.ttl;
+
+/**
+ * This class uses the Datastax driver to read/write locators from
+ * Cassandra metrics_locator Column Family.
+ */
+public class DatastaxLocatorIO implements LocatorIO {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DatastaxLocatorIO.class);
+
+    public static final String KEY = "key";
+    public static final String COLUMN1 = "column1";
+    public static final String VALUE = "value";
+
+    private PreparedStatement getValue;
+    private PreparedStatement putValue;
+
+    public DatastaxLocatorIO() {
+        createPreparedStatements();
+    }
+
+    /**
+     * Create all prepared statements use in this class for metrics_locator
+     */
+    private void createPreparedStatements()  {
+
+        // create a generic select statement for retrieving from metrics_locator
+        Select.Where select = QueryBuilder
+                .select()
+                .all()
+                .from( CassandraModel.CF_METRICS_LOCATOR_NAME )
+                .where( eq ( "key", bindMarker() ));
+        getValue = DatastaxIO.getSession().prepare( select );
+
+        // create a generic insert statement for inserting into metrics_locator
+        Insert insert = QueryBuilder.insertInto(CassandraModel.QUOTED_KEYSPACE, CassandraModel.CF_METRICS_LOCATOR_NAME)
+                .using(ttl(TenantTtlProvider.LOCATOR_TTL))
+                .value(KEY, bindMarker())
+                .value(COLUMN1, bindMarker())
+                .value(VALUE, bindMarker());
+        putValue = DatastaxIO.getSession()
+                .prepare(insert)
+                .setConsistencyLevel( ConsistencyLevel.ONE);  // TODO: remove later; required by the cassandra-maven-plugin 2.0.0-1
+                                                              // (see https://issues.apache.org/jira/browse/CASSANDRA-6238)
+
+    }
+
+    /**
+     * Insert a locator with key = shard long value calculated using Util.getShard()
+     * @param locator
+     * @throws IOException
+     */
+    @Override
+    public void insertLocator(Locator locator) throws IOException {
+        Session session = DatastaxIO.getSession();
```

Comment:
There was no instrumentation fir insertLocator in the original code, so I didn't add it here.  Don't want to instrument/send more things to graphite if we are not tracking it.